library_registry/admin/controller.py

```diff
@@ -42,9 +50,54 @@ def log_in(self):
         else:
             return INVALID_CREDENTIALS
 
+    def log_in_with_token(self):
+        """JWT Token based login to create more RESTful API
+
+        Returns:
+            Response: With valid credentials this will return JWT Access and Refresh tokens as Set-Cookie headers and redirects to 'admin/admin_view
+            Response: With invalid credentials this will return the INVALID CREDENTIALS response
+        """
+        username = request.form.get("username")
+        password = request.form.get("password")
+        if Admin.authenticate(self._db, username, password):
+            access_token = create_access_token(identity=username)
+            refresh_token = create_refresh_token(identity=username)
+            response = make_response(
+                redirect(url_for('admin.admin_view')), 302)
+            set_access_cookies(response, access_token)
+            set_refresh_cookies(response, refresh_token)
+            return response
+        else:
+            return INVALID_CREDENTIALS
+
+    def refresh_token(self):
```

Comment:
If our clients use cookies, they are going to expect cookie-like behavior. This refresh setup requires setting up some checks on the frontend to check if an access token is expired and then redirect to the refresh endpoint to get the new access token. Normal cookie behavior would not require this redirect and instead would update the Expire attribute, so I'm not sure how a client using cookies would make it to the refresh endpoint without also including unexpected logic on the frontend. Have you looked at [implicit refresh with cookies](https://flask-jwt-extended.readthedocs.io/en/stable/refreshing_tokens/#implicit-refreshing-with-cookies)? To support cookies, let's modify this and surrounding code to mimic expected cookie behavior and then retest.

Reply:
Implemented.